ladybug/color.py

```diff
@@ -488,12 +561,20 @@ def __getitem__(self, key):
 
     def __setitem__(self, key, value):
         """Set a color to a new color in color list."""
-        self._colors[key] = value
-
-    def __delitem__(self, key):
-        """Remove a color from the color list."""
-        del self._colors[key]
+        assert isinstance(value, Color), 'Expected ladybug color. ' \
+            'Got {}'.format(type(value))
+        _new_cols = list(self._colors)
+        _new_cols[key] = value
+        self._colors = tuple(_new_cols)
```

Comment:
Are we not worried about the case where someone appends or deletes a color from the list and then breaks the ColorRange because we were not able to perform a check to make sure the number of colors aligns with the number of domain items (what happens now in the `colors.setter`)? I understand why we want to avoid casting everything to a tuple in some cases but the color range gets used so infrequently that it seems it is worth an extra casting for the peace of mind that the `colros` property is valid.

Reply:
Makes sense but in that case do we really need to support item assignment in the first place?

Reply:
Very fair point.  I had kinda felt the `__setitem__` here was a bit forced.  I will take it out.